### PR TITLE
DOMA-7937 marketplace roles

### PR DESCRIPTION
**Pull request by nomerdvadcatpyat**
- Added `canReadMarketplace`, `canManageMarketplace`, `canReadPaymentsWithInvoices` permissions
- Set marketplace permissions to existed roles (according to [doc](https://docs.google.com/spreadsheets/d/1_j5p1fEmrs2NXPgrCNYKlZ1UzJLxft-soA6yFTwhI8c/edit#gid=0))
- Slpit `Payment` access to `canReadPayments` and `canReadPaymentsWithInvoices`
- Fixed some bugs

TODO:

- [x] After `AcquiringIntegrationContext` is updated with fields associated with invoice, divide the accesses to `AcquiringIntegrationContext` by `canReadMarketplace/canManageMarketplace`

### CODE REVIEW

**Review by VKislov — APPROVED** [review]

File: apps/condo/domains/acquiring/access/Payment.js [hunk 1]
```diff
@@ -77,7 +93,10 @@ async function canReadPaymentsSensitiveData ({ authentication: { item: user }, e
     }
 
     // Otherwise check if it's employee or not
-    return !!(await checkOrganizationPermission(user.id, existingItem.organization, 'canReadPayments'))
+    const canReadPayments = !!(await checkOrganizationPermission(user.id, existingItem.organization, 'canReadPayments'))
+    const canReadPaymentsWithInvoices = !!(await checkOrganizationPermission(user.id, existingItem.organization, 'canReadPaymentsWithInvoices'))
+
+    return canReadPayments || canReadPaymentsWithInvoices
```
Comment (Alllex202) [inline apps/condo/domains/acquiring/access/Payment.js:99]: there is no need to do a second check if the first one is successful

File: apps/condo/domains/marketplace/schema/Invoice.js
Comment (Alllex202) [inline apps/condo/domains/marketplace/schema/Invoice.js:1]: Are there any tests that check this behavior?

File: apps/condo/domains/marketplace/schema/MarketItem.js [hunk 2]
```diff
@@ -46,6 +50,24 @@ const MarketItem = new GQLListSchema('MarketItem', {
         },
 
     },
+    hooks: {
+        afterChange: async ({ context, operation, originalInput, existingItem }) => {
+            const isSoftDeleteOperation = operation === 'update' && existingItem && isSoftDelete(originalInput)
```
Comment (Alllex202) [inline apps/condo/domains/marketplace/schema/MarketItem.js:55]: some tests?

File: apps/condo/domains/organization/constants/common.js [hunk 3]
```diff
@@ -247,6 +254,15 @@ const DEFAULT_ROLES = {
...
         'canReadServices': true,
+        'canReadPaymentsWithInvoices': false,
+        'canReadMarketplace': false,
+        'canManageMarketplace': false,
+        'canReadInvoices': true,
+        'canManageInvoices': false,
+        'canReadMarketItemPrices': true,
+        'canReadMarketPriceScopes': true,
+        'canReadMarketItems': true,
+        'canManageMarketItems': false,
```
Comment (Alllex202) [inline apps/condo/domains/organization/constants/common.js:265]: Foreman can read Market Items and cannot Read Marketplace? Is everything correct here?

File: apps/condo/migrations/20231213105633-0352_auto_20231213_0556.js [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+--
+ALTER TABLE "OrganizationEmployeeRoleHistoryRecord" ADD COLUMN "canReadMarketplace" boolean NULL;
+--
+-- Add field canReadPaymentsWithInvoices to organizationemployeerolehistoryrecord
+--
+ALTER TABLE "OrganizationEmployeeRoleHistoryRecord" ADD COLUMN "canReadPaymentsWithInvoices" boolean NULL;
+
+--
+-- [CUSTOM] Set marketplace permissions to existed roles
+--
```
Comment (Alllex202) [inline apps/condo/migrations/20231213105633-0352_auto_20231213_0556.js:42]: What changes are there for the Master role?

Comment (nomerdvadcatpyat) [reply apps/condo/migrations/20231213105633-0352_auto_20231213_0556.js:42]: `employee.role.Foreman.name`
str number 66

File: apps/condo/domains/organization/constants/common.js (hunk 3 above)
Comment (nomerdvadcatpyat) [reply apps/condo/domains/organization/constants/common.js:265]: fixed it

File: apps/condo/migrations/20231213105633-0352_auto_20231213_0556.js (hunk 4 above)
Comment (Alllex202) [reply apps/condo/migrations/20231213105633-0352_auto_20231213_0556.js:42]: oops, oh that English :)

File: apps/condo/domains/marketplace/schema/Invoice.js
Comment (nomerdvadcatpyat) [reply apps/condo/domains/marketplace/schema/Invoice.js:1]: Invoice.test.js validation block

File: apps/condo/domains/marketplace/schema/MarketItem.js (hunk 2 above)
Comment (nomerdvadcatpyat) [reply apps/condo/domains/marketplace/schema/MarketItem.js:55]: Added

File: apps/condo/domains/acquiring/access/Payment.js [hunk 5]
```diff
@@ -77,7 +93,15 @@ async function canReadPaymentsSensitiveData ({ authentication: { item: user }, e
...
-    return !!(await checkOrganizationPermission(user.id, existingItem.organization, 'canReadPayments'))
+    const canReadPayments = !!(await checkOrganizationPermission(user.id, existingItem.organization, 'canReadPayments'))
+    if (canReadPayments) {
+        return true
+    }
+
+    const canReadPaymentsWithInvoices = !!(await checkOrganizationPermission(user.id, existingItem.organization, 'canReadPaymentsWithInvoices'))
+    if (canReadPaymentsWithInvoices) {
+        return true
+    }
```
Comment (Alllex202) [inline apps/condo/domains/acquiring/access/Payment.js:104]:
```suggestion
    return canReadPaymentsWithInvoices
```

**Review by Alllex202 — APPROVED** [review]

**Pull request merged**